### PR TITLE
Add cookbook team / release process docs

### PR DESCRIPTION
**Pull request by tas50**
Also format the others and improve several areas that were confusing or no longer accurate.

### CODE REVIEW

File: TESTING.MD [hunk 1]
```diff
@@ -30,74 +24,159 @@ rake style:ruby:auto_correct  # Auto-correct RuboCop offenses
...
+The first part is straight forward. The travis gem can encrypt environment variables against the public key on the Travis repository and add them to the .travis.yml.
+
+```
+        > Format markdown
+```
 
 If you are a Chef employee, you will need to get VMWare Fusion, [Vagrant plugin for VMWare](https://www.vagrantup.com/vmware/), and set up a [Hashicorp Atlas](https://www.hashicorp.com/atlas.html) account that has access to the Chef images.
 
-Once you have the required software, login to Atlas from the command line. Uncomment the suites in .kitchen.yml file, or set the environment variable KITCHEN_LOCAL_YAML to kitchen.vmware.yml.
+<<<<<<< a2522b87718a266938e656c7597c34e6fa1571f3 Once you have the required software, login to Atlas from the command line. Uncomment the suites in .kitchen.yml file, or set the environment variable KITCHEN_LOCAL_YAML to kitchen.vmware.yml.
```
Comment (iennae) [inline TESTING.MD]: Is this "<<<<<<< a2522b87718a266938e656c7597c34e6fa1571f3 " actually in the text? That looks like some meta information accidentally getting into documentation.

File: CONTRIBUTING.MD [hunk 2]
```diff
@@ -78,58 +55,21 @@ Date:   Wed Sep 18 11:44:40 2015 -0700
...
-
-
-### Expected Result:
-[What are you expecting to happen as the consequence of above reproduction steps?]
-
-
-### Actual Result:
-[What actually happens after the reproduction steps?]
-```
+Chef uses Github Issues to track issues with the community cookbooks. Issues should be submitted in the appropriate cookbook's Github repository at: <https://github.com/chef-cookbooks/COOKBOOK_NAME/issues/new>.
```
Comment (iennae) [inline CONTRIBUTING.MD]: Instead of "Chef" "Chef community cookbook maintainers"

File: CONTRIBUTING.MD [hunk 3]
```diff
@@ -142,86 +82,53 @@ contributors understand the purpose of changes to the code.
...
-changes don't cause regressions for other use cases.  Ideally all
-changes include either functional or unit tests.  See the TESTING.md
-file for additional information on testing in chef cookbooks.
+Chef cookbooks are tested with functional and unit tests to ensure changes don't cause regressions for other use cases. Ideally all changes include either functional or unit tests. See the TESTING.md file for additional information on testing in chef cookbooks.
 
 ## Code Review
 
-Chef Software regularly reviews code contributions and provides suggestions
-for improvement in the code itself or the implementation.
+Chef Software regularly reviews code contributions and provides suggestions for improvement in the code itself or the implementation.
```
Comment (iennae) [inline CONTRIBUTING.MD]: Instead of Chef Software, "Chef community cookbook maintainers"

File: CONTRIBUTING.MD [hunk 4]
```diff
@@ -142,86 +82,53 @@ contributors understand the purpose of changes to the code.
...
+
+Please do indicate new platform (families) or platform versions in the commit message, and update the relevant ticket.
+
+If a contribution adds new platforms or platform versions, indicate such in the body of the commit message(s), and update the relevant issues. When writing commit messages, it is helpful for others if you indicate the issue. For example:    git commit -m '[ISSUE-1041] - Updated pool resource to correctly delete.'
+
+Please do ensure that your changes do not break or modify behavior for other platforms supported by the cookbook. For example if your changes are for Debian, make sure that they do not break on CentOS.
+
+Please do **not** modify the version number in the `metadata.rb`, Chef will select the appropriate version based on the release cycle information above.
```
Comment (iennae) [inline CONTRIBUTING.MD]: Instead of "Chef" ""Chef community cookbook maintainers"

File: CONTRIBUTING.MD [hunk 5]
```diff
@@ -142,86 +82,53 @@ contributors understand the purpose of changes to the code.
...
+
+Please do **not** update the `CHANGELOG.md` for a new version. Not all changes to a cookbook may be merged and released in the same versions. Chef Software will update the `CHANGELOG.md` when releasing a new version of the cookbook.
```
Comment (iennae) [inline CONTRIBUTING.MD]: Instead of "Chef Software" "Chef community cookbook maintainers"

File: RELEASE_PROCESS.MD [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+## Follow Semver
+
+Infrastructure stability for our users is critical so we aim to release predictably versioned cookbooks. Following Semver allows users to pin cookbook requirements in wrapper cookbooks or environment files, and avoid breaking changes. See <http://semver.org/>
+
+## Test Before Releasing
+
+Many cookbooks have integration tests that run using kitchen-docker/kitchen-dokken in Travis CI.  For these cookbooks we're preforming integration testing on every commit. For others integration tests must be run locally and this should be done before any release to ensure high quality releases.
+
+## Always update the changelog
```
Comment (iennae) [inline RELEASE_PROCESS.MD:17]: We want to be clear here that we don't mean for folks contributing to do this. :) we say don't do this, and here we are saying "Always update"

File: TESTING.MD [hunk 7]
```diff
@@ -1,22 +1,16 @@
...
+# Cookbook TESTING doc
 
-Testing Prerequisites
----------------------
-Chef cookbooks require either a working ChefDK installation set as your system's default ruby or Ruby 2.0+ with bundler installed. Using ChefDK provides a consistent Ruby install, and is the suggested method. ChefDK can be downloaded at https://downloads.chef.io/chef-dk/
+## Testing Prerequisites
+
+Chef cookbooks require either a working ChefDK installation set as your system's default ruby or Ruby 2.0+ with bundler installed. Using ChefDK provides a consistent Ruby install, and is the suggested method. ChefDK can be downloaded at <<https://downloads.chef.io/chef-dk/>
 
 To ensure all gems are updated to their latest releases run `bundle install; bundle update` before running any Rake testing tasks.
```
Comment (iennae) [inline TESTING.MD:7]: Do we want people to do bundle anything when developing now? I thought we were anti-recommending this?

File: TESTING.MD [hunk 8]
```diff
@@ -30,74 +24,159 @@ rake style:ruby:auto_correct  # Auto-correct RuboCop offenses
...
 or
-```
+
+```shell
 rake integration:vagrant
 ```
 
-Private Images
-------------------
+<<<<<<< a2522b87718a266938e656c7597c34e6fa1571f3
```
Comment (iennae) [inline TESTING.MD]: This looks like another artifact that got pulled in.

File: TESTING.MD (hunk 7 above)
Comment (tas50) [inline TESTING.MD:7]: This stuff wasn't really updated by this PR and mostly needs to get rewritten.  It's some legacy stuff with modifications.

**Pull request merged**